### PR TITLE
Feat(parser): add create user sql parser. 

### PR DESCRIPTION
**Pull request by dongzl**
### What is changed and how it works?

ref #61  

What's Changed:

### Related changes

- PR to update `pisanix/docs/cn`:
- PR to update `pisanix/protocol`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- add create user sql parser.
- add create user sql parser test case.

There is a grammar conflict, but I can't solve it. If someone can fix it, we can discuss it.

```
/**
      TODO: There is a grammar conflict, but I can't solve it.
            If someone can fix it, we can discuss it.
    | user identified_with_plugin opt_initial_auth
      {
           UserWithAuthOption::UserIdentifiedWithPlugin(UserIdentifiedWithPlugin {
               span: $span,
               user: $1,
               identified_with_plugin: $2,
               opt_initial_auth: $3,
           })
      }
    */
```

### CODE REVIEW

File: pisa-proxy/parser/mysql/src/grammar.y [hunk 1]
```diff
@@ -7302,6 +7353,488 @@ alter_lock_option_value -> String:
...
+      }
+    | user identified_with_plugin opt_initial_auth
+      {
+           UserWithAuthOption::UserIdentifiedWithPlugin(UserIdentifiedWithPlugin {
+               span: $span,
+               user: $1,
+               identified_with_plugin: $2,
+               opt_initial_auth: $3,
+           })
+      }
```
Comment (dongzl) [inline pisa-proxy/parser/mysql/src/grammar.y:7386]: There is conflict for `user identification opt_create_user_with_mfa` and `user identified_with_plugin opt_initial_auth`.

If remove the `user identified_with_plugin opt_initial_auth`, the test case at `parser#182` is OK.

and how to fix it.

**Comment by dongzl** [comment]
```
/Users/dongzonglei/.cargo/bin/cargo test --message-format=json-diagnostic-rendered-ansi --color=always --no-run --lib parser::test::test_dal_stmt --manifest-path /Users/dongzonglei/source_code/Github/pisanix/pisa-proxy/parser/mysql/Cargo.toml
   Compiling mysql_parser v0.1.0 (/Users/dongzonglei/source_code/Github/pisanix/pisa-proxy/parser/mysql)
error: failed to run custom build command for `mysql_parser v0.1.0 (/Users/dongzonglei/source_code/Github/pisanix/pisa-proxy/parser/mysql)`
Caused by:
  process didn't exit successfully: `/Users/dongzonglei/source_code/Github/pisanix/pisa-proxy/target/debug/build/mysql_parser-ca2195c3cbb66656/build-script-build` (exit status: 1)
  --- stderr
  Error: CTConflictsError{12 Reduce/Reduce, 1 Shift/Reduce}
Process finished with exit code 101
```

**Comment by xuanyuan300** [comment]
I will try to reslove it.

**Comment by xuanyuan300** [comment]
This is a reduce/reduce conflicts, When parsing the rule `identified_with_plugin`, it has two cases as follows:
1. Reduce to `user identified_with_plugin  opt_initial_auth` rule.
2. Reduce to `user identification opt_create_user_with_mfa` rule.
So, we need to redesign rule to avoid ambiguity.

**Review by xuanyuan300 — APPROVED** [review]

**Comment by xuanyuan300** [comment]
LGTM

**Pull request merged**